temporian/implementation/numpy_cc/operators/window.cc

```diff
@@ -38,18 +38,36 @@ py::array_t<OUTPUT> accumulate(const ArrayD &evset_timestamps,
   size_t begin_idx = 0;
 
   for (size_t sampling_idx = 0; sampling_idx < n_event; sampling_idx++) {
+    // Note: We accumulate values in (t-window_length, t] with t=
+    // v_timestamps[sampling_idx], and there may be several contiguous equal
+    // values in v_timestamps.
 
+    // Add all values with same timestamp as the current one.
     accumulator.Add(v_values[sampling_idx]);
+    size_t same_ts_idx = sampling_idx;
+    while (
+      same_ts_idx + 1 < n_event &&
+      v_timestamps[same_ts_idx] == v_timestamps[same_ts_idx + 1]

```

Comment:
What about storing this value before the while loop on line 48. e.g.,

const auto ts = v_timestamps[same_ts_idx];
while (.... && ts == v_timestamps[same_ts_idx + 1]) { ...}